python/lsst/pipe/tasks/insertFakeSources.py

```diff
@@ -0,0 +1,597 @@
+        band : `str`
+        photoCalib : `lsst.afw.image.calib.Calib` or `lsst.afw.image.photoCalib.PhotoCalib`
+        psf : `lsst.meas.extensions.psfex.psfexPsf.PsfexPsf`
+
+        Returns
+        -------
+        starImages : `list`
+            A list of `lsst.afw.image.image.image.ImageF` of fake stars
+        """
+
+        starImages = []
+
+        self.log.info("Making %d fake star images" % len(fakeCat))
+
+        for (index, row) in fakeCat.iterrows():
+            # jointCal photoCalib and normal photoCalib have different ways of doing this.
+            xy = afwGeom.Point2D(row["x"], row["y"])
+            if self.config.useUpdatedCalibs and self.calexpType == "calexp":
+                # This should go away when this is switched to reference catalogs
+                try:
+                    flux = photoCalib.magnitudeToInstFlux(row[band + "magVar"], xy)
+                except LogicError:
+                    flux = 0
+            else:
+                flux = photoCalib.getFlux(row[band + "magVar"])
+
+            try:
+                starIm = psf.computeImage(xy)
+
+            except InvalidParameterError:
+                self.log.info("Star at %0.4f, %0.4f outside of image" % (row["x"], row["y"]))
+                continue
+
+            starIm *= flux
+            starImages.append((starIm.convertF(), xy))
+
+        return starImages
+
+    def cleanCat(self, fakeCat):
+        """Remove rows from the fakes catalog which have HLR = 0 for either the buldge or disk component
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+
+        Returns
+        -------
+        fakeCat : `pandas.core.frame.DataFrame`
+        """
+
+        goodRows = ((fakeCat[self.config.bulgeHLR] != 0.0) & (fakeCat[self.config.diskHLR] != 0.0))
+
+        badRows = len(fakeCat) - len(goodRows)
+        self.log.info("Removing %d rows with HLR = 0 for either the bulge or disk" % badRows)
+
+        return fakeCat[goodRows]
+
+    def addFakeSources(self, calexp, fakeImages, sourceType):
+        """Add the fake sources to the given calexp
+
+        Parameters
+        ----------
+        fakeCat : `pandas.core.frame.DataFrame`
+        calexp : `lsst.afw.image.exposure.exposure.ExposureF`
+        fakeImages : `list`
+            A list of `lsst.afw.image.image.image.ImageF`
+
+        Returns
+        -------
+        calexp : `lsst.afw.image.exposure.exposure.ExposureF`
+
+        Notes
+        -----
+        Uses the x, y information in the ``fakeCat`` to position an image of the fake interpolated onto the
+        pixel grid of the image. Sets the ``FAKE`` mask plane for the pixels added with the fake source.
+        """
+
+        calexpBBox = calexp.getBBox()
+        calexpMI = calexp.maskedImage
+
+        for (fakeImage, xy) in fakeImages:
+            X0 = xy.getX() - fakeImage.getWidth()/2 + 0.5
+            Y0 = xy.getY() - fakeImage.getHeight()/2 + 0.5
```

Comment:
I can never keep track of if our indexing is at the corner of a pixel or the center and frequently get it wrong. You may be 100% right here, just wanted to bring it up in case you wanted to double check like I always need to.

Reply:
I did check